TODO: I should probably use see: `man program_invocation_name` and `program_invocation_short_name`
see misc/init-misc.c in glibc

this does work however since I'm also getting full args, maybe I should keep full path too! doneTODO: just set __progname = __progname_full and that's all!

as to why I'm doing this, see comment in patch: /home/user/build/1packages/4used/glibc/lame-o-chromium_setproctitle_spotted.patch


--- a/services/service_manager/embedder/set_process_title_linux.cc	2019-06-06 20:51:32.000000000 +0200
+++ b/services/service_manager/embedder/set_process_title_linux.cc	2019-06-18 17:19:33.896991832 +0200
@@ -47,6 +47,8 @@
 #include <unistd.h>
 
 extern char** environ;
+extern char *__progname; //in glibc's misc/init-misc.c
+extern char *__progname_full;
 
 static char** g_main_argv = NULL;
 static char* g_orig_argv0 = NULL;
@@ -100,6 +102,14 @@ void setproctitle(const char* fmt, ...)
   }
   va_end(ap);
   g_main_argv[1] = NULL;
+  if (__progname_full) {
+    //char *p = strrchr (__progname_full, '/');
+    //if (p == NULL) {
+      __progname = __progname_full;
+    //} else {
+    //  __progname = p + 1;
+    //}
+  }
 }
 
 // A version of this built into glibc would not need this function, since
